### PR TITLE
Issue #11783: prepare-settings.sh fails to create settings.xml

### LINKED ISSUE #11783: prepare-settings.sh fails to create settings.xml
```
✔ ~/java/github/checkstyle/checkstyle [master|✔] 
$ ./.ci/prepare-settings.sh 
cp: cannot create regular file '~/.m2/settings.xml': No such file or directory

```

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Resolves #11783

### CODE REVIEW

File: .ci/prepare-settings.sh [hunk 1]
```diff
@@ -27,12 +27,13 @@ replace GPG_PASSPHRASE
...
+TEMP_SETTING=./.ci-temp/release-settings.xml
+SETTING=~/.m2/settings.xml
 
 if cmp -s "$TEMP_SETTING" "$SETTING"; then
-  TODAY=$(date + "%y%m%d")
-  mv $SETTING $SETTING.backup."${TODAY}"
+  TODAY=$( date '+%y%m%d' )
+  mv "$SETTING" "$SETTING".backup."${TODAY}"
+  cp "$TEMP_SETTING" "$SETTING".backup
+  else
```
Comment (romani) [inline .ci/prepare-settings.sh:37]: Fix indentation

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/prepare-settings.sh [hunk 2]
```diff
@@ -27,12 +27,13 @@ replace GPG_PASSPHRASE
...
-TEMP_SETTING="./.ci-temp/release-settings.xml"
-SETTING="~/.m2/settings.xml"
+TEMP_SETTING=./.ci-temp/release-settings.xml
+SETTING=~/.m2/settings.xml
 
 if cmp -s "$TEMP_SETTING" "$SETTING"; then
-  TODAY=$(date + "%y%m%d")
-  mv $SETTING $SETTING.backup."${TODAY}"
+  TODAY=$( date '+%y%m%d' )
+  mv "$SETTING" "$SETTING".backup."${TODAY}"
```
Comment (pbludov) [inline .ci/prepare-settings.sh:35]: Better to make it single string
`"$SETTING".backup."${TODAY}"` => `"${SETTING}.backup.${TODAY}"`

**Review by pbludov — CHANGES_REQUESTED** [review]

File: .ci/prepare-settings.sh (hunk 1 above)
Comment (romani) [reply .ci/prepare-settings.sh:39]: done

File: .ci/prepare-settings.sh (hunk 2 above)
Comment (romani) [reply .ci/prepare-settings.sh]: done

**Comment by romani** [comment]
testing:
```
✔ ~/java/github/romani/checkstyle [Rahulkhinchi03/setting-2 L|✚ 1] 
$ source ~/.m2/export-cs-release-vars.sh 

✔ ~/java/github/romani/checkstyle [Rahulkhinchi03/setting-2 L|✚ 1] 
$ head -n 4 ~/.m2/settings.xml
<settings>
  <servers>
    <server> <!-- -->
      <id>sonatype-nexus-staging</id>

✔ ~/java/github/romani/checkstyle [Rahulkhinchi03/setting-2 L|✚ 1] 
$ ./.ci/prepare-settings.sh 
file /home/rivanov/.m2/settings.xml already exists
backup /home/rivanov/.m2/settings.xml to /home/rivanov/.m2/settings.xml.backup.20220703070304

✔ ~/java/github/romani/checkstyle [Rahulkhinchi03/setting-2 L|✚ 1] 
$ ./.ci/prepare-settings.sh 
file /home/rivanov/.m2/settings.xml already exists
no backup is required as content is same

```

**Review by romani — APPROVED** [review]
ok to merge

**Review by rnveach — APPROVED** [review]

**Pull request merged**